setup.sh

```diff
@@ -0,0 +1,234 @@
+      git reset --hard origin/develop
+    elif [ "1" == $(git branch -vv |grep -c "* master") ]
+    then
+      echo "- Hard resetting of 'master' branch."
+      git reset --hard origin/master
+    fi
+else
+    echo "Reset ignored because you are not on 'master' or 'develop'."
+fi
+
+python3.6 -m venv .env
+updateenv
+}
+
+function config_generator () {
+
+echo "Starting to generate config.json"
+
+echo "
+-------------------------
+General configuration
+-------------------------
+"
+read -p "Max open trades: (Default: 3)
+" max_trades
+
+read -p "Stake amount: (Default: 0.05)
+" stake_amount
+
+read -p "Stake currency: (Default: BTC)
+" stake_currency
+
+read -p "Fiat currency: (Default: USD)
+" fiat_currency
+
+echo "
+------------------------
+Bittrex config generator
+------------------------
+"
+read -p "Exchange API key
+" api_key
+read -p "Exchange API Secret
+" -s api_secret
+
+echo "
+-------------------------
+Telegram config generator
+-------------------------
+"
+read -p "Telegram Token
+" -s token
+read -p "Telegram Chat_id
+" chat_id
+
+sed -e "s/\"max_open_trades\": 3,/\"max_open_trades\": $max_trades,/g" \
+    -e "s/\"stake_amount\": 0.05,/\"stake_amount\": $stake_amount,/g" \
+    -e "s/\"stake_currency\": \"BTC\",/\"stake_currency\": \"$stake_currency\",/g" \
+    -e "s/\"fiat_display_currency\": \"USD\",/\"fiat_display_currency\": \"$fiat_currency\",/g" \
+    -e "s/\"your_echange_key\"/\"$api_key\"/g" \
+    -e "s/\"your_echange_secret\"/\"$api_secret\"/g" \
+    -e "s/\"your_instagram_token\"/\"$token\"/g" \
+    -e "s/\"your_instagram_chat_id\"/\"$chat_id\"/g"
+    -e "s/\"dry_run\": false,/\"dry_run\": true,/g" config.json.example > config.json
+
+}
+
+function config () {
+go_to_folder
+if [ -f config.json ]
+then
+read -p "A config file already exist, do you want to override it [Y/N]?
+"
+if [[ $REPLY =~ ^[Yy]$ ]]
+then
+    config_generator
+else
+    echo "Configuration of config.json ignored."
+fi
+else
+    config_generator
+fi
+
+echo "Edit ./config.json to modify Pair and other configurations."
+}
+
+function install () {
+echo "
+-------------------------
+Install mandatory dependencies
+-------------------------
+"
+
+go_to_folder
+if [ "$(uname -s)" == "Darwin" ]
+then
+    echo "- You are on macOS"
+    install_macos
+elif [ -x "$(command -v apt-get)" ]
+then
+    echo "- You are on Debian/Ubuntu"
+    install_debian
+else
+    echo "This script does not support your OS."
+    echo "If you have Python3.6, pip, virtualenv, ta-lib you can continue."
+    echo "Wait 10 seconds to continue the next install steps or use ctrl+c to interrupt this shell."
+    sleep 10
+fi
+reset
+echo "
+- Install complete.
+"
+config
+echo "You can now use the bot by executing 'source .env/bin/activate; python3 freqtrade/main.py'."
+}
+
+function plot () {
+echo "
+-----------------------------------------
+Install dependencies for Plotting scripts
+-----------------------------------------
+"
+pip install plotly
```

Comment:
you could also use `pip install plotly --upgrade` if the user has a old installation